src/utils/customHooks.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+
+export const useShowAndHideAnimation = (
+  blockOneInView: boolean,
+  // add correct type, AnimationControls was suggested
+  animation: any,
+  // add correct type, AnimationControls was suggested
+  textAnimation: any
```

Comment:
I committed this custom hook kind of, please try to find the correct types here I did not have time to research but AnimationControls was the suggested one.

Reply:
Not been able to find much info online on this, but it is the suggested type for me as well, and Typescript is happy.